webapp/blueprint.py

```diff
@@ -7,11 +7,40 @@
     "gui", __name__, template_folder="/templates", static_folder="/static"
 )
 
+JAAS_URL = "https://jaas.ai"
+INDEX = "index.html"
+
+
+def loggedIn():
+    return "loggedin" in flask.request.cookies
```

Comment:
It occurs to me, the most important audience for these redirects are people who are logged in *currently*, and would expect to still see the GUI after the switch-over. Those people won't have a `loggedin` cookie, as we will have only just released this code, but they will have an `auth_tkt` cookie. So I think we should probably check for the existence of either. Do you agree?

(Also, I think it's likely to surprise people less if you treat `loggedin` as an actual boolean value - `return flask.request.cookies.get('loggedin') == 'true'` or something)

Reply:
I had originally thought about including the `auth_tkt` but decided against it because I am trying to avoid adding in legacy patches for something that is only a minor speedbump for the actively logged in users. Users get logged out all the time :) 

I can add this back in if you feel strongly about it.

I've updated the cookie value check to be more explicit in the meantime.

Reply:
I'm worried about seriously confusing existing users. Currently. how used do you think people are to going to https://jujucharms.com (rather than https://jujucharms.com/new) to access the GUI?

We're changing the structure here quite a bit. On the day we switch over https://jujucharms.com to point to this version of the app, anyone who has been used to going to https://jujucharms.com to access the GUI will now go to https://jaas.ai, and the "log in" button will have disappeared. I think that's significantly jarring, don't you?

I agree that preserving `auth_tkt` is legacy mess, which is unfortunate, but it seems to me that the better user experience *vastly* outweighs that particular downside (as it usually does). Our first responsibility is to the users. Arguable this whole redirect system is legacy mess anyway - one more cookie name in our conditional is the least of our problems =)